.github/workflows/maven.yml

```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        jdk: [8, 11]
+        jdk: [17]
```

Comment:
do we actually want to set Java 17 as minimum?

Reply:
Good question. Spring boot 3 requires Java 17. However, some other modules could work with older versions

Reply:
it's actually hard to say if and how many people are using togglz with legacy Java. I have actually no clue about the different use cases of the togglz users.

@MediaMarco uses togglz with Spring Boot. Many others as well. But not sure about the distribution of number of Spring Boot users and number of all togglz users.. :/ 

I just checked it. Quarkus is also on Java 17. So I think we should just got for it. 
Even if I would open a poll or something like that, I would not get enough responses to know if it would be a show stopper to set java 17 as minimum. Let's just do it and see if someone actually needs it to be on a older Java version or not.